### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by chris-ha458**
Some of these deserve explanations :
c7c8f22
- I apologize for the churn here. I changed this just a PR ago, but i forgot about this method. Reading the documentation for the [method](https://doc.rust-lang.org/std/primitive.str.html#method.lines) leads me to believe that this is the most correct approach. If I understand the code correctly, this library does not need any special line handling, just simple proper line handling that hopefully works cross platform. This seems to be exactly that.

2479d5f
- This was how i debugged c7c8f22 for my platform, and it seems better to have an `expect` rather than a naked `unwrap`. As explained above, this part is just simple file handling of a file with a well known format so it seem like more complex error handling is unnecessary.

28974cb
- when if blocks contain control statements like `return` `break` `continue`, `else` statements can often safely be removed.
- Whether the match based version is clearer is unclear. However if it is not, an multiple if based version is included in the [comments](https://github.com/progval/unicode_names2/commit/28974cb2a72f714fc808b9bf8331faab0583669a#commitcomment-129761599).  
- Feel free to lmk if you'd prefer the if based version.

### CODE REVIEW

File: generator/src/util.rs [hunk 1]
```diff
@@ -30,29 +30,30 @@ pub struct Split<'a, 'b> {
...
-                return Some(ret);
+                // trailing data
+                self.done = true;
+                Some(self.s)
             }
         }
-        // trailing data
-        self.done = true;
-        Some(self.s)
     }
```
Comment (progval) [inline generator/src/util.rs]: I'm not convinced this is more readable than the existing code. Could you revert the change?

Comment (chris-ha458) [reply generator/src/util.rs]: Would you prefer to `else` as well?
If i take them out the first part would look like this
```
        if self.done {
            return None;
        } 
        if self.s.is_empty() {
            self.done = true;
            return Some("");
        } 
...
```
The reason I ask is that if-else with returns in them triggers some lints.

Of course if you'd rather keep them in, regardless, I'll do that.

**Comment by chris-ha458** [comment]
reverted and removed unnecessary if's
468e4e2

File: generator/src/util.rs (hunk 1 above)
Comment (progval) [reply generator/src/util.rs]: Those sound like quite opinionated lints then. I'm usually fond of functional style, but this is a bit too much IMO.
